api/rest/restapi.go

```diff
@@ -647,12 +647,27 @@ func (api *API) unpinHandler(w http.ResponseWriter, r *http.Request) {
 }
 
 func (api *API) allocationsHandler(w http.ResponseWriter, r *http.Request) {
+	queryValues := r.URL.Query()
+	all := queryValues.Get("a")
 	var pins []types.PinSerial
-	err := api.rpcClient.Call("",
-		"Cluster",
-		"Pins",
-		struct{}{},
-		&pins)
+	var err error
+	if all == "true" {
+		err = api.rpcClient.Call(
+			"",
+			"Cluster",
+			"Pins",
+			struct{}{},
+			&pins,
+		)
+	} else {
+		err = api.rpcClient.Call(
+			"",
+			"Cluster",
+			"PinsQuiet",
```

Comment:
Handle all pintype filtering here in the REST API. For the moment I want this to be an API feature because it is solely a presentation helper (I could perfectly live without it) and I don't want it to change internal APIs just because the user wants less things on the screen.

When the time comes that our states are so big that this becomes a feature for efficiency, then it should be addressed in the state component direclty and easily removed from here.